tdms/src/openandorder.cpp

```diff
@@ -29,7 +31,7 @@ void openandorder(const char *mat_filename, char **matrix_names, const mxArray *
   //fprintf(stderr, "Got all %d matrices\n",nmatrices);
   //fprintf(stderr, "%d\n",mxGetFieldNumber( (mxArray *)matrixptrs[0], "I_tot"));
 
-  if (!are_equal(matrix_names[0], "fdtdgrid")){
+  if (matrix_names[0] != "fdtdgrid"){
```

Comment:
Ah, of course this doesn't work if `"fdtdgrid"` is the _last_ entry in the vector.
I'm going to leave this up to confirm the build fails before fixing.

Reply:
[Very good](https://github.com/UCL/TDMS/actions/runs/3750619750/jobs/6370566162).